docs/content/guides/dagster/branch-depl-with-snowflake.mdx

```diff
@@ -0,0 +1,348 @@
+
+- `ITEMS` - A table containing the entire dataset
+- `COMMENTS` - A table containing data about comments
+- `STORIES` - A table containing data about stories
+
+To set up a branch deployment workflow to construct and test these tables, we will:
+
+1. Define these tables as [assets](/concepts/assets/software-defined-assets).
+2. Configure our assets to write to Snowflake using a different connection (credentials and database name) for two environments: production and branch deployment.
+3. Clone our production database upon each branch deployment launch, naming each clone `PRODUCTION_CLONE_<ID>`, where `<ID>` is the pull request ID of the branch. Then we'll create a branch deployment and test our Hacker News assets against our newly cloned database.
+4. Delete the corresponding database clone upon closing the feature branch.
+
+---
+
+## Step 1: Create our job
+
+In production, we want to write three tables to Snowflake:`ITEMS`, `COMMENTS`, and `STORIES`. We can define these tables as [assets](/concepts/assets/software-defined-assets) as follows:
+
+```python file=/guides/dagster/local_to_production/assets.py startafter=start_assets endbefore=end_assets
+# assets.py
+import pandas as pd
+import requests
+
+from dagster import asset
+
+
+@asset(
+    config_schema={"N": int},
+    io_manager_key="snowflake_io_manager",
+)
+def items(context) -> pd.DataFrame:
+    """Items from the Hacker News API: each is a story or a comment on a story."""
+    rows = []
+    max_id = requests.get(
+        "https://hacker-news.firebaseio.com/v0/maxitem.json", timeout=5
+    ).json()
+    # Hacker News API is 1-indexed, so adjust range by 1
+    for item_id in range(max_id - context.op_config["N"] + 1, max_id + 1):
+        item_url = f"https://hacker-news.firebaseio.com/v0/item/{item_id}.json"
+        rows.append(requests.get(item_url, timeout=5).json())
+
+    # ITEM_FIELD_NAMES is a list of the column names in the Hacker News dataset
+    result = pd.DataFrame(rows, columns=ITEM_FIELD_NAMES).drop_duplicates(subset=["id"])
+    result.rename(columns={"by": "user_id"}, inplace=True)
+    return result
+
+
+@asset(
+    io_manager_key="snowflake_io_manager",
+)
+def comments(items: pd.DataFrame) -> pd.DataFrame:
+    """Comments from the Hacker News API."""
+    return items[items["type"] == "comment"]
+
+
+@asset(
+    io_manager_key="snowflake_io_manager",
+)
+def stories(items: pd.DataFrame) -> pd.DataFrame:
+    """Stories from the Hacker News API."""
+    return items[items["type"] == "story"]
+```
+
+As you can see, our assets use a [I/O manager](/concepts/io-management/io-managers) named `snowflake_io_manager` (source code available in the [Transitioning Data Pipelines from Development to Production](/guides/dagster/local-to-production-tutorial#part-one-local-development) guide). Using I/O managers and other resources allow us to swap out implementations per environment without modifying our business logic.
```

Comment:
```suggestion
As you can see, our assets use an [I/O manager](/concepts/io-management/io-managers) named `snowflake_io_manager`. Using I/O managers and other resources allow us to swap out implementations per environment without modifying our business logic.
```
the snowflake IO manager source code is in the dagster-snowflake library, not in the local to prod guide